platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/properties/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,200 @@
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpringAndClassLoader() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpringAndClassLoader(
+            PROPERTIES_FILENAME, this.getClass().getClassLoader());
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpringAndClassLoaderFileDNE() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpringAndClassLoader(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithFileSystem() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithFileSystem(propertiesFile.getPath(), null);
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithFileSystemFileDNE() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithFileSystem(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadAsResource() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadAsResource("/" + PROPERTIES_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadAsResourceFileDNE() throws Exception {
+    Properties testProperties = PropertiesLoader.attemptLoadAsResource(NON_EXISTENT_FILENAME, null);
+
+    Assert.assertEquals(
+        "Returns empty properties", Collections.emptySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testSubstituteSystemPropertyPlaceholders() throws Exception {
+    // set up map to replace system properties
+    Map<Object, Object> testMapSystemProperties = new HashMap<>();
+    testMapSystemProperties.put("Java Home", "${java.home}");
+    testMapSystemProperties.put("Java Version", "${java.version}");
+
+    // set up mock and insert into method
+    when(propertiesMock.entrySet()).thenReturn(testMapSystemProperties.entrySet());
+    Properties testProperties =
+        PropertiesLoader.substituteSystemPropertyPlaceholders(propertiesMock);
+
+    // manually replace system properties
+    testMapSystemProperties.replace("Java Home", System.getProperty("java.home"));
+    testMapSystemProperties.replace("Java Version", System.getProperty("java.version"));
```

Comment:
❗️ Instead of mutating state that you've already passed, which makes this difficult to follow and verify what is being tested, you should have two separate maps. One can be your expected value. One can be the value you pass with the mock.